aea/protocols/base.py

```diff
@@ -43,9 +43,28 @@ def __init__(self, body: Optional[Dict] = None,
         :param body: the dictionary of values to hold.
         :param kwargs: any additional value to add to the body. It will overwrite the body values.
         """
+        self._counterparty = None  # type: Optional[Address]
         self._body = copy(body) if body else {}  # type: Dict[str, Any]
         self._body.update(kwargs)
 
+    @property
+    def counterparty(self) -> Address:
+        """
+        Get the counterparty of the message in Address form.
+
+        :return the address
+        """
+        assert self._counterparty is not None
```

Comment:
Provide a statement next to it: "Counterparty not set!"

Reply:
+1